apps/base-docs/base-camp/docs/frontend-setup/building-an-onchain-app.md

```diff
@@ -0,0 +1,202 @@
+:::
+
+### Connectors
+
+Next, configure the chains, wallet connectors, and providers for your app. We'll use the `publicProvider` for now, to get started. See our guide on [Connecting to the Blockchain] for more information on blockchain providers.
+
+```typescript
+const { chains, publicClient } = configureChains([mainnet, base, baseGoerli], [publicProvider()]);
+```
+
+The default connectors provided by RainbowKit automatically enable the most popular wallets, so we'll add that next:
+
+```typescript
+const { connectors } = getDefaultWallets({
+  appName: 'My RainbowKit App',
+  projectId: 'YOUR_PROJECT_ID',
+  chains,
+});
+```
+
+You'll need a `projectId` from [Wallet Connect Cloud], which you can get for free on their site. Make sure to insert it in the appropriate place.
+
+```danger
+Remember, everything on the frontend is public!  Be sure to configure the allowlist for your WalletConnect id!
+```
+
+### Config
+
+Finally, add the config for wagmi:
+
+```typescript
+const wagmiConfig = createConfig({
+  autoConnect: true,
+  connectors,
+  publicClient,
+});
+```
+
+Setting `autoConnect` to `true` will allow your app to automatically reconnect users once they connect for the first time. Most people want this.
+
+## Wrapping Context Providers
+
+You can now wrap your app with the context providers for RainbowKit and wagmi. This will make your connection to the blockchain available throughout your entire app without needing to pass anything through props.
+
+```typescript
+export default function RootLayout({ children }: { children: React.ReactNode }) {
+  return (
+    <html lang="en">
+      <WagmiConfig config={wagmiConfig}>
+        <RainbowKitProvider chains={chains}>
+          <body className={inter.className}>{children}</body>
+        </RainbowKitProvider>
+      </WagmiConfig>
+    </html>
+  );
+}
+```
+
+## Adding the Connect Button
+
+You're now ready to add your connect button. You can do this anywhere in your app, thanks to the `RainbowKitProvider`. Common practice would be to place the button in your app's header. Since the Next.js template doesn't have one, we'll just add it to the top of the automatically generated page, rather than spending time implementing React components.
+
+Open up `index.tsx`, and import the `ConnectButton`:
+
+```typescript
+import { ConnectButton } from '@rainbow-me/rainbowkit';
+```
+
+Then, simply add the `ConnectButton` component at the top of the first `<div>`:
+
+```typescript
+// This function has been simplified to save space.
+export default function Home() {
+  return (
+    <main
+      className={`flex min-h-screen flex-col items-center justify-between p-24 ${inter.className}`}
+    >
+      <div className="z-10 w-full max-w-5xl items-center justify-between font-mono text-sm lg:flex">
+        <ConnectButton />
```

Comment:
This is the simple approach, should we leave a note also on where to learn for customizing the Button? https://www.rainbowkit.com/docs/custom-connect-button